Auth: Add `No basic role` UI

linoman opened the pull request:
**What is this feature?**

1. Adds tooltips to the OrgRole menu
2. Allows to choose the `No basic role` option for invite org users and 

**Why do we need this feature?**

Full details can be found [here](https://github.com/grafana/grafana-authnz-team/issues/134)

**Who is this feature for?**

AuthNZ

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-authnz-team/issues/256

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

linoman committed:
Add tooltip support for RadioButton component

linoman committed:
Change layout of RadioButton options

linoman committed:
Render OrgRole options

linoman committed:
Allow RoleOrgPicker for a horizontal layout

alexanderzobnin reviewed (changes requested):
Those changes in grafana/ui RadioButton group break everything in the UI:

![Screenshot from 2023-08-01 11-11-05](https://github.com/grafana/grafana/assets/4932851/73fa1a68-a884-4a29-b620-be44ba5a5b2f)

Also, we don't need this since we have `<RadioButtonList />` component
https://developers.grafana.com/ui/latest/index.html?path=/story/forms-radiobuttonlist--controlled-component

linoman committed:
Revert changes to ui components

linoman committed:
Addres PR comments

linoman committed:
Revert changes from UI components

linoman committed:
Revert UI component changes

Clarity-89 commented on packages/grafana-ui/src/components/Forms/Field.tsx:
```diff
@@ -30,6 +32,8 @@ export interface FieldProps extends HTMLAttributes<HTMLDivElement> {
   /** make validation message overflow horizontally. Prevents pushing out adjacent inline components */
   validationMessageHorizontalOverflow?: boolean;
+  /** Adds tooltip next to the label */
+  tooltipMessage?: string;
```
For the horizontal form components we use `description` prop to provide extra context, maybe you could use it instead of adding the tooltip? I don't think we use tooltips in horizontal form components and only use them in the inline form components in the design system, but maybe @JoaoSilvaGrafana can confirm.

JoaoSilvaGrafana replied:
Yes preferably we would not add a tooltip to `Field` without a reasonable check on how this would work fully. It would need to be documented on when you use tooltip vs description and what does this mean for all the other places that use description, if some of them would be changed to tooltip or not. So if we could use description here it would be best

alexanderzobnin reviewed (changes requested):
Few issues I observed:
1. Basic role is not selectable. Clicking on role does not change anything and just closes menu.
2. On the invite page in OSS there's a _No basic role_ option, however it's not listed in the role selector. I guess it should not be possible to use empty role in OSS. Anyway, it should be aligned and either listed everywhere in OSS or hidden.

linoman commented:
@alexanderzobnin 
> Basic role is not selectable. Clicking on role does not change anything and just closes menu.
This is very interesting as the suggested `RadioButtonList` doesn't behave similarly to the `RadioButtonGroup`. Thank you for pointing it out, I'll work on this.

linoman committed:
Revert changes to UI component

linoman committed:
Fix buggy RadioButtonList

linoman committed:
Move Tooltip from UI component to opts in Label

linoman committed:
Change question marks to information symbol

linoman committed:
Fix text alignment

linoman commented:
I have addressed the comments and requested changes

@alexanderzobnin Thank you for helping me out with figuring out what happened to the RadioButtonList component. I applied your suggested change and now it works as expected.

@grafana/grafana-frontend-platform I have reverted all the UI changes. I'd appreciate a review from your end 🙏🏻

alexanderzobnin reviewed (commented):
Looks good. The only thing left is invite form issue in OSS.

linoman committed:
Add flag for enterprise only feature

linoman committed:
Use licensedAccessControlEnabled
